src/libraries/System.Text.Json/gen/Reflection/TypeExtensions.cs

```diff
@@ -109,6 +109,30 @@ public static bool IsNullableValueType(this Type type, out Type? underlyingType)
             return false;
         }
 
+        public static bool CanContainNullableReferenceTypeAnnotations(this Type type)
+        {
+            // Returns true iff Type instance has potential for receiving nullable reference type annotations,
+            // i.e. the type is a reference type or contains generic parameters that are reference types.
+
+            if (!type.IsValueType)
```

Comment:
I haven't read the whole PR but make sure this works with `Nullable<T>`

Reply:
Good point. `Nullable<T>` types may or may not admit nullable annotation, for example `(int, int)?` cannot contain nullable reference types whereas `(string, string)?` can. I'll add a property to the test class that validates this.